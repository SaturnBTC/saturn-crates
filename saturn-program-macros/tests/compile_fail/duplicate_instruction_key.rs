use saturn_program_macros::saturn_program;

mod instruction {
    #[derive(borsh::BorshSerialize, borsh::BorshDeserialize)]
    pub enum Instr {
        Dummy,
    }
}

#[saturn_program(
    instruction = "crate::instruction::Instr",
    instruction = "crate::instruction::Instr"
)]
mod handlers {}

fn main() {} 