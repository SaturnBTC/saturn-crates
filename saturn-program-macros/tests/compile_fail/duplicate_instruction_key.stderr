error: duplicate `instruction` key; specify it only once
  --> tests/compile_fail/duplicate_instruction_key.rs:12:5
   |
12 |     instruction = "crate::instruction::Instr"
   |     ^^^^^^^^^^^
